### PR TITLE
Add stats function to get total size of each message within log

### PR DESCRIPTION
**Pull request by IamPete1**
Lets us do things like this:
newplot (23)

Handy for working out what you should turn off if you want to make your logs smaller.

### CODE REVIEW

File: parser.js [hunk 1]
```diff
@@ -903,6 +903,62 @@ class DataflashParser {
...
+
+                const msg_size = get_msg_size(msg.Format)
+                const size = msg_size * count
+                ret[msg.Name] = { count, msg_size, size }
+
+                total_size_check += size
+            }
+        }
+
+        // Total should match log size
```
Comment (IamPete1) [inline parser.js:941]: This is only really a sanity check, we could remove it. Or add the difference as an item "corrupted".

Comment (Williangalvani) [reply parser.js:941]: I thnk we should do a warn instead of log. but I think it makes sense to keep it.

File: parser.js [hunk 2]
```diff
@@ -903,6 +903,62 @@ class DataflashParser {
...
+        // Need this to access get_size_of in the helper function
+        const get_size_of_fun = this.get_size_of
+        function get_msg_size(Format) {
+            // Work out size of single message, all message have a 3 byte header
+            let size = 3
+            for (let i = 0; i < Format.length; i++) {
+                size += get_size_of_fun(Format.charAt(i))
+            }
+            return size
+        }
```
Comment (Williangalvani) [inline parser.js]:
```suggestion
        get_msg_size(Format) => {
            // Work out size of single message, all message have a 3 byte header
            let size = 3
            for (let i = 0; i < Format.length; i++) {
                size += this.get_size_of(Format.charAt(i))
            }
            return size
        }
```

Arrow functions automatically bind to `this` from the outer scope

File: parser.js (hunk 1 above)
Comment (IamPete1) [reply parser.js:941]: Changed to warn and removed the early return. So you get the warning but you still get some stats.

File: parser.js (hunk 2 above)
Comment (IamPete1) [reply parser.js]: Thanks, done.

**Review by Williangalvani — APPROVED** [review]
nice, thanks!

**Pull request merged**